core/src/main/java/com/linecorp/armeria/server/http/dynamic/ParamMap.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.dynamic;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Pre-defined interface type for the whole request parameter names and values.
+ */
+public interface ParamMap extends Map<String, List<String>> {
+
+    /**
+     * Return the first value for the given key.
+     *
+     * @param key the key to find
+     * @return the first value for the specified key, or null
+     */
+    default String getFirst(String key) {
+        List<String> values = get(key);
+        if (values != null && !values.isEmpty()) {
+            return values.get(0);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Return the first value for the given key, or
+     * {@code defaultValue} if this map contains no mapping for the key.
+     *
+     * @param key he key to find
+     * @param defaultValue the default mapping of the key
+     * @return the first value for the given key, or
+     *         {@code defaultValue} if this map contains no mapping for the key
+     */
+    default String getFirstOrDefault(String key, String defaultValue) {
```

Comment:
I also wrote `getFirst()` at first but changed to the same notation of `Map.getOrDefault`.  I also prefer shorter one. Will fix!